sphinx/domains/std.py

```diff
@@ -747,3 +728,33 @@ def get_objects(self):
     def get_type_name(self, type, primary=False):
         # never prepend "Default"
         return type.lname
+
+    def is_enumerable_node(self, node):
+        return node.__class__ in self.enumerable_nodes
```

Comment:
Do you want to avoid derived enumerable nodes?
If we accept derived nodes, isinstance is good solution.
```python
return isinstance(node, self.enumerable_nodes.keys())
```